src/Perlang.Interpreter/Resolution/ResolveError.cs

```diff
@@ -2,6 +2,9 @@
 
 namespace Perlang.Interpreter.Resolution
 {
+    // TODO: Rename to NameResolutionError, but need to resolve the conflict with existing type. Could we rename that
+    // one to NameResolutionTypeValidationError, or perhaps just moving it to the Typing namespace is the right thing to
+    // do?
```

Comment:
This TODO comment must be fixed properly before merge.

Reply:
This is currently being dealt with as part of fixing this, separate PR coming up.

> The end result was pretty good and this should be ready for merge, but I might break out parts of to separate PR:s to keep history cleaner.

Reply:
> separate PR coming up.

Was merged as #211, resolving.